### PR TITLE
Issue 4214 - Custom Ticket:Groups Views input upgrade UI

### LINKED ISSUE #4214: Custom Tickets: Groups - Views input upgrade UI
### Description
For accessing the new groups the view input will be updated to accomodate the groups. 
**Important** : when updating the view now it creates groups with the whole state of the viewer (overrides and hidden groups), the camera and clipping planes are still saved in the camera prop

### Goals
- [ ] To have a new cleaner ui for views
- [ ] To have a way to access and change groups

### Tasks
- [ ] Update the ui with the new design
- [ ] Add a way to navigate to the groups card inside of the ticket card 

### Related Resources
[MockUp](https://www.figma.com/file/3FjNEOdQbcHwzquVZxUqMl/3D-Repo?node-id=3042-23547&t=wExZ71I9NEiTL5BX-0)

### PR DESCRIPTION
**Pull request by Amantini1997**
This fixes #4214 

#### Description
Updated UI

#### Test cases
Go to a custom ticket with a view property

### CODE REVIEW

**Comment by The-Daniel** [comment]
image
A few questions about how this works. 
Are groups only available if you have at least one viewpoint property on a ticket?
If you have more than one viewpoint will these all go to the same groups window?

File: frontend/src/v5/ui/controls/actionMenu/actionMenuItem/actionMenuItem.component.tsx [hunk 1]
```diff
@@ -14,20 +14,30 @@
...
-		{children}
-	</MenuItem>
-);
-ActionMenuItem.isActionMenuClosingElement = true;
+}: ActionMenuItemProps) => {
+	const { close } = useContext(ActionMenuContext);
+
+	const handleClick = (e) => {
+		onClick?.(e);
+		if (!disabled) {
```
Comment (The-Daniel) [inline frontend/src/v5/ui/controls/actionMenu/actionMenuItem/actionMenuItem.component.tsx:37]: Disabled seems to be broken in a lot of places.
image
With viewer permissions the Viewpoints ellipses appears disabled but can be opened (it also still has the tooltip). 'Create viewpoint' isn't disabled thus gives you an error modal upon clicking.

I don't think the ellipses icon should be disabled if a viewpoint has been set, as a viewer should be able to 'Go to viewpoint'. The create and delete options should be disabled though
image

The view/image options use the ellipses menu item components instead of this action menu component. The ellipses menu items don't have this `if (!disabled)` clause so clicking a disabled menu item closes the menu (which I think it shouldn't?)

Also, perhaps the option for groups should say "View groups" for user permissions that can view but not edit.
image

File: frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/properties/basicTicketImage/basicTicketImage.component.tsx [hunk 2]
```diff
@@ -47,15 +47,17 @@ export const BasicTicketImage = ({
...
-				<TicketImageDisplayer
-					imgSrc={value}
-					disabled={disabled || !isProjectAdmin}
-					onEmptyImageClick={uploadImage}
-				/>
+				<Label>{label}</Label>
+				<Content>
+					<TicketImageDisplayer
+						imgSrc={value}
+						disabled={disabled || !isProjectAdmin}
```
Comment (The-Daniel) [inline frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/properties/basicTicketImage/basicTicketImage.component.tsx:54]: This disabled value doesn't work. (I have viewer permissions)
image
The cursor doesn't change, but I can still click on it to open a file explorer window. Also should we remove the "Add image" text if the user doesn't have permissions?

You can just add `pointer-events: none;` to the styled component when disabled

**Review by The-Daniel — CHANGES_REQUESTED** [review]

File: frontend/src/v5/ui/controls/actionMenu/actionMenu.component.tsx [hunk 3]
```diff
@@ -14,62 +14,51 @@
...
 	TriggerButton: any;
 	PopoverProps?: any;
+	onOpen?: (e) => void;
+	onClose?: (e) => void;
 };
-
 export const ActionMenu = ({
-	className,
-	TriggerButton: TriggerButtonInput,
+	TriggerButton,
```
[the remark this answers is not in the document]
Comment (Amantini1997) [reply frontend/src/v5/ui/controls/actionMenu/actionMenu.component.tsx:31]: I actually removed it by mistake. Thanks for spotting this!

File: frontend/src/v5/ui/controls/ellipsisMenu/ellipsisMenu.component.tsx [hunk 4]
```diff
@@ -15,78 +15,34 @@
...
-			handleCloseDropdown();
-		}
-	};
-
-	return (
-		<>
+export const EllipsisMenu = ({ selected, children, className, disabled }: IEllipsisMenu): JSX.Element => (
+	<ActionMenu
+		TriggerButton={(
 			<Tooltip title={formatMessage({ id: 'ellipsisMenu.tooltip', defaultMessage: 'More options' })}>
```
[the remark this answers is not in the document]
Comment (Amantini1997) [reply frontend/src/v5/ui/controls/ellipsisMenu/ellipsisMenu.component.tsx:34]: I agree with you, but I think it's like a convention. You could easily refer to the component as the "more options button" and people will still get it. Maybe @george-hadfield knows this better though

**Comment by sanmont3drepo** [comment]
This bit of functionality is missing
image

**Review by The-Daniel — CHANGES_REQUESTED** [review]

File: frontend/src/v5/ui/controls/ellipsisMenu/ellipsisMenu.component.tsx [hunk 5]
```diff
@@ -15,78 +15,44 @@
...
-	const handleListKeyDown = (event) => {
-		if (event.key === 'Tab') {
-			event.preventDefault();
-			handleCloseDropdown();
-		}
-	};
+export const EllipsisMenu = ({ selected, children, className, disabled }: IEllipsisMenu): JSX.Element => {
+	if (disabled) {
+		return (
+			<EllipsisButton variant={selected ? 'secondary' : 'primary'} disabled={disabled}>
```
[the remark this answers is not in the document]
Comment (Amantini1997) [reply frontend/src/v5/ui/controls/ellipsisMenu/ellipsisMenu.component.tsx:34]: Yeah, I'd rather have the `disabled` option on the tooltip instead of the empty string too tbh. We could do that, but I would not that in this branch as I feel a few components could use the new approach.

The reason why I created the if is both the tooltip and the actionMenu, which in the first case is not required.
I think this is cleaner.

**Comment by george-hadfield** [comment]
### Increase height of accordion in groups by around 5px
Looks smaller in height than design

image

**Comment by george-hadfield** [comment]
### Reduce size of blue circle and number to match design in accordion headers

This doesn't seem to match the ratio used in the design. Tried playing around with the css but had no luck in changing values. also increase the space slightly between title and the left of the blue circle as currently looks a bit too tight.

image

**Comment by george-hadfield** [comment]
### Closed groups should use the outline version of the circle and chevron
This is to replicate other areas within io that this is used

image

**Comment by george-hadfield** [comment]
### Change checkbox hover state
This should simply change the border colour to primary.main on hover

See prototype for hover state interaction: https://www.figma.com/proto/bRqIl0g9psqD3ISkmxwcNp/3D-Repo?page-id=2628%3A17653&type=design&node-id=3251-35436&viewport=-43868%2C-6127%2C1&scaling=min-zoom&starting-point-node-id=3612%3A33527

image

**Comment by george-hadfield** [comment]
### Add hover states to the group item icons that appear on hover

See prototype for hover interaction: https://www.figma.com/proto/bRqIl0g9psqD3ISkmxwcNp/3D-Repo?page-id=2628%3A17653&type=design&node-id=3251-35436&viewport=-43868%2C-6127%2C1&scaling=min-zoom&starting-point-node-id=3612%3A33527

image

**Comment by george-hadfield** [comment]
### Scrollbar only needed if characters/lines exceed the height of what's visible

image

**Review by george-hadfield — CHANGES_REQUESTED** [review]
Feedback left

**Comment by Amantini1997** [comment]
> ### Selecting a parent item should also select all the children of that selection
> Currently just shows the one you selected as checked
> 
> image

This is not part of this branch as that is a logic functionality that cannot be implemented at the moment, so will be fixed in another branch, later on

**Comment by Amantini1997** [comment]
\azp run

**Comment by george-hadfield** [comment]
### Change text colour of number inside circle to primary.main

image

**Comment by george-hadfield** [comment]
### Change plus to tick icon for saving default view
Also move the tooltip closer to the icon

image

**Comment by george-hadfield** [comment]
### Change weight of collection titles to 500
If this is an already defined style. Please change globally.

**Comment by george-hadfield** [comment]
### Add tooltip to target icon
"Go to default view"

image

**Review by george-hadfield — APPROVED** [review]

**Review by The-Daniel — CHANGES_REQUESTED** [review]

File: frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/properties/ticketView/ticketView.component.tsx
[the remark this answers is not in the document]
Comment (Amantini1997) [reply frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/properties/ticketView/ticketView.component.tsx:1]: As mentioned earlier, this has to do with the lack of logic underneath

File: frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/properties/ticketView/ticketView.styles.ts [hunk 6]
```diff
@@ -14,49 +14,47 @@
...
+	${/* sc-selector */ EllipsisButton}:hover {
+		background-color: transparent;
+	}
 `;
+
+export const Tooltip = styled(TooltipBase).attrs({
+	componentsProps: {
+		popper: {
+			sx: {
+				marginLeft: '-10px !important',
```
[the remark this answers is not in the document]
Comment (Amantini1997) [reply frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/properties/ticketView/ticketView.styles.ts:56]: I am not that concerned about a warning, but let's see if I can get rid of that

File: frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/properties/ticketView/ticketView.component.tsx [hunk 7]
```diff
@@ -50,72 +55,179 @@ export const TicketView = ({
...
-					<TicketImageAction disabled={disabled}>
-						<EditViewpointIcon />
-						<FormattedMessage id="viewer.card.ticketView.action.editViewpoint" defaultMessage="Edit viewpoint" />
-					</TicketImageAction>
-				)}
+		<InputContainer disabled={disabled} {...props}>
+			<Header>
+				<Label>{label}</Label>
+				<HeaderSection>
+					{!hasViewpoint ? (
```
[the remark this answers is not in the document]
Comment (Amantini1997) [reply frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/properties/ticketView/ticketView.component.tsx:120]: This has to do with the lack of logic underneath. See the next comment where this is explained better

**Review by The-Daniel — APPROVED** [review]

**Comment by sanmont3drepo** [comment]
Can you integrate the current update and gotoViewpoint with this UI?

**Review by sanmont3drepo — CHANGES_REQUESTED** [review]

File: frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/ticketGroups/groupsAccordion/groupsAccordion.component.tsx [hunk 8]
```diff
@@ -21,15 +21,16 @@ import { IGroup } from '@/v5/store/tickets/groups/ticketGroups.types';
...
 import { Groups } from '../groups/groups.component';
 import { AddGroupButton } from '../groups/groupActionMenu/addGroupButton/addGroupButton.component';
 
-type GroupsAccordionProps = {
+type GroupsAccordionProps = Omit<AccordionProps, 'Icon' | 'children'> & {
 	title: any;
 	groups: IGroup[];
 };
-export const GroupsAccordion = ({ title, groups = [] }: GroupsAccordionProps) => {
+export const GroupsAccordion = ({ title, groups = [], ...props }: GroupsAccordionProps) => {
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/ticketGroups/groupsAccordion/groupsAccordion.component.tsx:33]: Why this change?

File: frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/properties/ticketView/ticketView.component.tsx [hunk 9]
```diff
@@ -70,59 +81,162 @@ export const TicketView = ({
...
+			<TicketImageContent
+				value={imgSrc}
+				onChange={onImageChange}
+				disabled={disabled}
+			>
+				{/* Image */}
+				<TicketImageActionMenu value={imgSrc} onChange={onImageChange} disabled={disabled} />
+				{/* Camera */}
+				<ViewActionMenu
+					disabled={!hasCamera}
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/properties/ticketView/ticketView.component.tsx:182]: Can we split all the actionMenus into different components like we did with ticketImageActionMenu.
Could be cameraActionMenu, groupsActionMenu

File: frontend/src/v5/ui/controls/actionMenu/actionMenu.component.tsx [hunk 10]
```diff
@@ -14,7 +14,7 @@
  *  You should have received a copy of the GNU Affero General Public License
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-import { ReactNode, useState } from 'react';
+import { ReactNode, useEffect, useState } from 'react';
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/controls/actionMenu/actionMenu.component.tsx:17]: Are the changes in this file still necessary?

**Review by sanmont3drepo — CHANGES_REQUESTED** [review]

File: frontend/src/v5/ui/controls/actionMenu/actionMenu.component.tsx (hunk 10 above)
Comment (Amantini1997) [reply frontend/src/v5/ui/controls/actionMenu/actionMenu.component.tsx]: Not anymore, I reverted them

**Review by sanmont3drepo — APPROVED** [review]
